cdap-tms/src/main/java/io/cdap/cdap/messaging/store/leveldb/LevelDBMessageTable.java

```diff
@@ -64,83 +73,102 @@ byte getType() {
     }
   }
 
-  private final DB levelDB;
-  private final TopicMetadata topicMetadata;
+  private final LevelDBPartitionManager partitionManager;
 
-  LevelDBMessageTable(DB levelDB, TopicMetadata topicMetadata) {
-    this.levelDB = levelDB;
-    this.topicMetadata = topicMetadata;
+  LevelDBMessageTable(LevelDBPartitionManager partitionManager) {
+    this.partitionManager = partitionManager;
   }
 
-  private void checkTopic(TopicId topicId, int generation) {
-    Preconditions.checkArgument(this.topicMetadata.getTopicId().equals(topicId), "Not allowed to use table with a " +
-      "different topic id. Table's topic Id: {}. Specified topic id: {}", this.topicMetadata.getTopicId(), topicId);
-    Preconditions.checkArgument(this.topicMetadata.getGeneration() == generation, "Not allowed to use table with " +
-                                  "a different generation id. Table's generation: {}. Specified generation: {}",
-                                this.topicMetadata.getGeneration(), generation);
+  @Override
+  protected CloseableIterator<RawMessageTableEntry> scan(ScanRequest scanRequest) throws IOException {
+    Collection<LevelDBPartition> partitions = partitionManager.getPartitions(scanRequest.getStartTime());
+    if (partitions.isEmpty()) {
+      return CloseableIterator.empty();
+    }
+    RawMessageTableEntry tableEntry = new RawMessageTableEntry();
+    BiFunction<byte[], byte[], RawMessageTableEntry> decodeFunction = (key, value) -> {
+      Map<String, byte[]> columns = decodeValue(value);
+      return tableEntry.set(MessageTableKey.fromBytes(key), columns.get(TX_COL), columns.get(PAYLOAD_COL));
```

Comment:
Can we reuse MessageTableKey instance instead of creating new one for each entry returned by the iterator?